modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/PollingSourceDeclarationEnricher.java

```diff
@@ -60,26 +71,47 @@ protected void onSource(SourceDeclaration source) {
 
             source.setRunsOnPrimaryNodeOnly(true);
 
-            ParameterDeclaration parameter = new ParameterDeclaration(SCHEDULING_STRATEGY_PARAMETER_NAME);
-            parameter.setDescription(SCHEDULING_STRATEGY_PARAMETER_DESCRIPTION);
-            parameter.setRequired(true);
-            parameter.setType(loader.load(SchedulingStrategy.class), false);
-            parameter.setExpressionSupport(NOT_SUPPORTED);
-            parameter.addModelProperty(new InfrastructureParameterModelProperty(10));
-            parameter.addModelProperty(new QNameModelProperty(new QName(CORE_NAMESPACE, SCHEDULING_STRATEGY_ELEMENT_IDENTIFIER,
-                                                                        CORE_PREFIX)));
-            parameter.setDslConfiguration(ParameterDslConfiguration.builder()
-                .allowsInlineDefinition(true)
-                .allowsReferences(false)
-                .allowTopLevelDefinition(false)
-                .build());
-
             thereArePollingSources.set(true);
 
-            source.getParameterGroup(DEFAULT_GROUP_NAME).addParameter(parameter);
+            source.getParameterGroup(DEFAULT_GROUP_NAME).addParameter(declareSchedulingStrategyParameter(loader));
+
+            if (isPollingSourceLimitEnabled(extensionLoadingContext)) {
+              source.getParameterGroup(DEFAULT_GROUP_NAME).addParameter(declarePollingSourceLimitParameter());
+            }
           }
         });
       }
+
+      private ParameterDeclaration declarePollingSourceLimitParameter() {
+        ParameterDeclaration parameter = new ParameterDeclaration(POLLING_SOURCE_LIMIT_PARAMETER_NAME);
+        parameter.setDescription(POLLING_SOURCE_LIMIT_PARAMETER_DESCRIPTION);
+        parameter.setRequired(false);
+        parameter.setType(BaseTypeBuilder.create(JAVA).numberType().integer().range(1, null).build(), false);
+        parameter.setExpressionSupport(NOT_SUPPORTED);
+        parameter.addModelProperty(new SyntheticModelModelProperty());
+        parameter.addModelProperty(new SinceMuleVersionModelProperty(POLLING_SOURCE_LIMIT_MULE_VERSION));
+        parameter.setLayoutModel(LayoutModel.builder().tabName(ADVANCED_TAB).build());
+
+        return parameter;
+      }
+
+      private ParameterDeclaration declareSchedulingStrategyParameter(ClassTypeLoader loader) {
+        ParameterDeclaration parameter = new ParameterDeclaration(SCHEDULING_STRATEGY_PARAMETER_NAME);
+        parameter.setDescription(SCHEDULING_STRATEGY_PARAMETER_DESCRIPTION);
+        parameter.setRequired(true);
+        parameter.setType(loader.load(SchedulingStrategy.class), false);
+        parameter.setExpressionSupport(NOT_SUPPORTED);
+        parameter.addModelProperty(new InfrastructureParameterModelProperty(10));
```

Comment:
I know this is unrelated to this change, but we need to review what's this sequence number for and why simply 10... These magic numbers are super fragile. Meanwhile, put this into a constant